packages/playwright/src/matchers/expect.ts

```diff
@@ -241,13 +252,28 @@ type ExpectMetaInfo = {
 
 class ExpectMetaInfoProxyHandler implements ProxyHandler<any> {
   private _info: ExpectMetaInfo;
+  private _prefixes: string[][];
 
-  constructor(info: ExpectMetaInfo) {
+  constructor(info: ExpectMetaInfo, prefixes: string[][]) {
     this._info = { ...info };
+    this._prefixes = prefixes;
   }
 
   get(target: Object, matcherName: string | symbol, receiver: any): any {
     let matcher = Reflect.get(target, matcherName, receiver);
+
+    if (typeof matcherName === 'string') {
+      for (const prefix of this._prefixes) {
+        for (let i = prefix.length; i > 0; i--) {
+          const qualifiedName = `${prefix.slice(0, i).join(':')}$${matcherName}`;
```

Comment:
It would be nice to extract this key generation since it is used in two distant places.